### PR TITLE
opae: add new opae-sdk spec to generate rpm

### PR DESCRIPTION
**Pull request by anandaravuri**
Fedora needs OPAE-SDK source code and OPAE-SDK spec file to generate rpms

Fedora Development environment

1) install dnf install fedora-packager @development-tools

2) build procedure in a clean chroot you need to configure your non-privileged account to be a member of the 'mock' group:
usermod -a -G mock <your username>

3) rpmdev-setuptree
sets up an RPM build area in your ~/rpmbuild directory. This directory will contain several subdirectories, for the project source code, RPM configuration files and for the resulting source and binary packages.

4) copy OPAE_SDK source code to ~/rpmbuild/SOURCE
cd ~/rpmbuild/SOURCES
Source code  opae-sdk-1.4.0.tar.gz

5)  The RPM package is configured by .spec files
copy opae-sdk-rpm.spec file to cd ~/rpmbuild/SPECS

6) Build rpm spec
rpmbuild -ba opae-sdk-rpm.spec

7) RRPM are in RPMS folder

example: https://fedoraproject.org/wiki/How_to_create_a_GNU_Hello_RPM_package

### CODE REVIEW

File: opae-sdk-rpm.spec [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+cd  mybuild
+make DESTDIR=%{buildroot} install
+mkdir -p %{buildroot}/etc/systemd/system/
+mv %{buildroot}%{_usr}/lib/systemd/system/fpgad.service  %{buildroot}/etc/systemd/system/fpgad.service
+
+%clean
+
+%post
+mkdir -p /etc/ld.so.conf.d
+echo "" > /etc/ld.so.conf.d/opae-c.conf
```
Comment (tswhison) [inline opae-sdk-rpm.spec]: opae-c.conf is populated by the ldconfig call?

File: opae-sdk-rpm.spec [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+BuildRequires:     gcc,gcc-c++
+BuildRequires:     cmake
+BuildRequires:     json-c-devel
+BuildRequires:     libuuid-devel
+BuildRequires:     rpm-build
+BuildRequires:     hwloc-devel
+BuildRequires:     gtest-devel
+BuildRequires:     python-sphinx
+BuildRequires:     doxygen
+
```
Comment (r-rojo) [inline opae-sdk-rpm.spec]: we still need runtime requires (json-c, uuid, ...)

File: opae-sdk-rpm.spec [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+%install
+cd  mybuild
+make DESTDIR=%{buildroot} install
+mkdir -p %{buildroot}/etc/systemd/system/
+mv %{buildroot}%{_usr}/lib/systemd/system/fpgad.service  %{buildroot}/etc/systemd/system/fpgad.service
+
+%clean
+
+%post
+mkdir -p /etc/ld.so.conf.d
```
Comment (r-rojo) [inline opae-sdk-rpm.spec]: use %{_sysconfdir} for /etc

**Comment by nakulkorde** [comment]
- Added helper script to eliminate manual steps.
- Updated specfile to rename the packages as ```opae-<package_name>-<version>-<release>.rpm

File: scripts/create-rpms.sh [hunk 4]
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
```
Comment (anandaravuri) [inline scripts/create-rpms.sh]: check for rpmbuild installed or not

File: scripts/create-rpms.sh [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+cd ~/rpmbuild/SPECS/
+
+#clean RPMS directory
+rm -rf ../RPMS/x86_64/opae-*
+
+#generate RPMS
+rpmbuild -ba opae-sdk-rpm.spec
+
+#copy RPMS to build directory
+cp ~/rpmbuild/RPMS/x86_64/opae-* $BUILD_DIR/
```
Comment (anandaravuri) [inline scripts/create-rpms.sh]: add rpmlint  and mock 
 rpmlint hello.spec ../SRPMS/hello* ../RPMS/*/hello*
 mock --verbose ../SRPMS/hello-2.10-1.fc25.src.rpm

File: scripts/create-rpms.sh (hunk 4 above)
Comment (tswhison) [reply scripts/create-rpms.sh]:
```bash
RPMBUILD=`which rpmbuild 2>/dev/null`
if [ "x${RPMBUILD}" = "x" ]; then
    printf "rpmbuild not found\n"
    exit 1
fi
```

File: opae-sdk-rpm.spec [hunk 6]
```diff
@@ -0,0 +1,231 @@
+Summary:        Open Programmable Acceleration Engine
+Name:           opae
+Version:        1.4.0
```
Comment (tswhison) [inline opae-sdk-rpm.spec]: 1.4.1

File: opae-sdk-rpm.spec (hunk 3 above)
Comment (anandaravuri) [reply opae-sdk-rpm.spec]: fixed

File: scripts/create-rpms.sh (hunk 4 above)
Comment (tswhison) [reply scripts/create-rpms.sh]: or that.. :)

File: scripts/create-rpms.sh (hunk 5 above)
Comment (nakulkorde) [reply scripts/create-rpms.sh]: I feel like these steps could be part of test rather than each build.

File: opae-sdk-rpm.spec (hunk 6 above)
Comment (nakulkorde) [reply opae-sdk-rpm.spec]: Currently we are at 1.4.0, will update to 1.4.1 when we do the next release. :)

**Pull request closed without merging**